test/test_common/simulated_time_system_test.cc

```diff
@@ -233,6 +233,12 @@ TEST_F(SimulatedTimeSystemTest, DuplicateTimer) {
   thread->join();
 }
 
+TEST_F(SimulatedTimeSystemTest, Enabled) {
+  TimerPtr timer = scheduler_->createTimer({});
+  timer->enableTimer(std::chrono::milliseconds(0));
+  EXPECT_TRUE(timer->enabled());
```

Comment:
I understand why it makes sense to check this behavior in simulated time, but I have a question: what production code would depend on this behavior in real time? Or is this aimed at testing code that would test for timer->enabled()?

Reply:
yeah, it's testing for code which checks timer->enabled().

I've been doing some timing fuzzing, where I have a transport socket wrapper which randomly fails to write, sets an alarm, and re-enabled writes.  It was breaking in a number of cases because of this (and some yet other undiscovered) bug.

Reply:
Just to make sure I understand: the code that checks timer->enabled() is test-only?

because that seems like an inherently racy code to run in production.

Reply:
The code that I'm encountering is test-only but we have production code that does the same.

Envoy is inherently siloed architecture.  If you arm a timer, it should be enabled until it is called.  All interactions between the timer for production Envoy are done in-thread, so there's really no race condition - this is totally a bug where simtime is not correctly implementing the timer API.